modules/renter/streamcache_test.go

```diff
@@ -102,6 +102,31 @@ func TestPruneCache(t *testing.T) {
 		t.Error("Cache size was changed by pruning to larger value")
 	}
 
+	// Confirm the same chunk won't be added if already added
+	id := sc.streamHeap[0].id
+	for i := 0; i < 5; i++ {
+		sc.Add(id, []byte{})
+	}
+
+	// Manually remove ID and confirm it no longer exists in the Heap of Cache
```

Comment:
I think it would be easier to just add the same id 5 times and check if the size of the heap and map increased by 1 each. That way you don't need to pop entries repeatedly and compare all the ids since you can be confident that it has only been added once anyway.